### DISCUSSION TITLE
How to bundle the json files that `lowdb` or `node-json-db` depends on?

### DISCUSSION THREAD

**Discussion by awdr74100** [yao-pkg/pkg#27] [Q&A] [answered]
Consider the following:

```json
{
  "pkg": {
    "assets": "./db.json",
    "targets": ["node20-win-x64"],
    "outputPath": "out"
  } 
}
```

The `db.json` file here is indeed bundled into the `.exe` file, and it is also normal to read and write through the code when running, I would like to ask whether this is a normal behavior? So the `db.json` file is bundled into the `.exe` file instead of possibly adding a file somewhere on the system and manipulating it.

Sorry for the weird question, but I do expect the `db.json` file to be bundled into the `.exe` file, and the `db.json` file in the `.exe` file to still be read and written, At first, I just wondered if the `db.json` in the `.exe` file could still be written to via `lowdb` and `node-json-db`? This means to perform normal fs module operations.

The question is whether the bundled json file can be written through a module such as fs? I thought that after all files were bundled into `.exe`, no writing modification operation could be carried out, please forgive me for the unfamiliarity of the binding behavior, thank you for anyone's help!

**Comment by robertsLando** [comment, marked as answer]
@awdr74100 If you need to read/write a file you need to have it outside the bundle. The alternative is to bundle it inside the exe, check if it's present in a defined "storage" directory by your choice and if not copy it there so you can read/write it as you prefer

**Comment by awdr74100** [comment]
@robertsLando Thank you very much for your quick reply! So I can put `db.json` in the user's home directory or a directory in the same location as the `.exe` file, and the bundled code will access the `db.json` in this location. Do I understand this correctly?

It seems that most applications are designed in this way. The `.exe` file is only used as a program to access external databases or resources. For example, `sqlite.db` or `lowdb.json` are external resources.

**Reply by robertsLando** [reply]
> @robertsLando Thank you very much for your quick reply! So I can put db.json in the user's home directory or a directory in the same location as the .exe file, and the bundled code will access the db.json in this location. Do I understand this correctly?

Yes exactly
